test/util/utils.js

```diff
@@ -251,6 +253,40 @@ function calculateContractAddress(senderAddress, senderNonce) {
   return getAddress(contract_address);
 }
 
+function calculateContractAddress2(senderAddress, cloneByteCodeHash, salt) {
+  const contract_address_long = solidityPackedKeccak256(
+    ["bytes1", "address", "bytes32", "bytes32"],
+    ["0xFF", senderAddress, salt, cloneByteCodeHash]
+  );
+
+  const contract_address = "0x" + contract_address_long.substring(26); //Trim the first 24 characters.
+
+  return getAddress(contract_address);
+}
+
+function getCloneByteCodeHash(beaconProxyAddress) {
+  return keccak256(
+    `0x3d602d80600a3d3981f3363d3d373d3d3d363d73${beaconProxyAddress.slice(2)}5af43d82803e903d91602b57fd5bf3`
+  );
+}
```

Comment:
where did you get this value? would be nice to have some comments here :)

Reply:
I got it from our contracts, but otherwise this is [EIP 1167](https://eips.ethereum.org/EIPS/eip-1167) style minimal clone.

We use the code from [this example](https://github.com/optionality/clone-factory/blob/master/contracts/CloneFactory.sol).

OZ uses the same: https://blog.openzeppelin.com/deep-dive-into-the-minimal-proxy-contract